tests/test_downloaders.py

```diff
@@ -449,7 +463,17 @@ def check_and_download_url_side_effect(url, download_dir, file_prefix='', **kwar
                 self.assertEqual(download_manager.download_dataset(dataset, ''), dataset_file_name)
                 self.assertTrue(logs_cm.records[0].message.startswith('Failed to download dataset'))
 
-    def test_download_dataset_failure(self):
+    def test_download_dataset_having_local_link_fails_because_of_corrupted_uri(self):
+        """Test that `download_dataset` raises a DownloadError exception if the download failed"""
+        download_manager = downloaders.DownloadManager()
+        dataset = Dataset.objects.get(pk=2)
+        with mock.patch.object(downloaders.HTTPDownloader, 'check_and_download_url') as mock_dl_url:
+            mock_dl_url.side_effect = downloaders.DownloadError
+            with self.assertRaises(downloaders.DownloadError):
+                with self.assertLogs(downloaders.LOGGER, logging.WARNING):
+                    download_manager.download_dataset(dataset, '')
+
+    def test_download_dataset_without_local_link_fails_because_of_corrupted_uri(self):
```

Comment:
The names you chose are confusing. Problems could happen for a number of reasons: an expired certificate, a wrong URI, the remote site is down for maintenance, etc... I think you should remove the `because_of_corrupted_uri` part.

Reply:
I have replaced `corrupted` with `unresponsive`.

Reply:
That is still too specific. As I wrote before, there can be multiple reasons for these errors.

Reply:
I remove the last part of the name